haystack/components/evaluators/results_evaluator.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any, Dict
+
+from pandas import DataFrame
+from pandas import concat as pd_concat
+
+
+class EvaluationResults:
+    def __init__(self, pipeline_name: str, results: Dict[str, Any]):
+        self.results = results
+        self.pipeline_name = pipeline_name
+
+    def score_report(self) -> DataFrame:
```

Comment:
I don't understand one thing. If the pipeline already returns aggregate scores, why do we re-compute them here?

Reply:
The proposal defined the input to initialise an instance of  `EvaluationResults` with individual score metrics:
https://github.com/deepset-ai/haystack/blob/main/proposals/text/7462-rag-evaluation.md

This was based on the this user case:

"My RAG pipeline has a low aggregate score, so I would like to see examples of specific inputs where the score was low to be able to diagnose what the issue could be."

We can also reiterate on this and include on the input besides individual score metric, aggregate metrics.

Reply:
My comment is about the following, which might be a misunderstanding on my part.

I thought the evaluators would return two fields: `individual_scores` and `scores`. The field `scores` would have the mean score for a pipeline, e.g. mean SAS score for the pipeline over all queries/answers. So my question was: why do we need to re-compute the mean of the individual scores in your function, if these means are already provided in the output of the evaluator?

If I understood correctly, what this function could do instead is just transforming that output to dataframe, so we have a function to get individual scores and transform that to dataframe, and another one to take aggregate/mean scores and transform that to dataframe.

I believe some alignment with @julian-risch on this is enough, so I will leave you to it.